Dart/src/com/jetbrains/lang/dart/ide/actions/DartPubActionBase.java

```diff
@@ -178,6 +178,8 @@ private static void doPerformPubAction(@NotNull final Module module,
 
     try {
       if (ourInProgress.compareAndSet(false, true)) {
+        command.withEnvironment("PUB_ENVIRONMENT", "IntelliJ-IDEA");
```

Comment:
@kevmoo, do we want a catch-all for anything IntelliJ, or more specific client ids?